cli/internal/terraform/terraform/aws-constellation/variables.tf

```diff
@@ -0,0 +1,110 @@
+variable "name" {
+  type        = string
+  description = "Name of your Constellation"
+  validation {
+    condition     = length(var.name) <= 10
+    error_message = "The length of the name of the Constellation must be <= 10 characters"
+  }
```

Comment:
Generally, having validation is always good, but I think it is sufficient to implement that kind of validation in the CLI here and have a single source of truth for what is valid. Failing earlier is good, but this will require us to update 2 places if we update our validation constraints.